src/lib/modules/scaffolding/index.ts

```diff
@@ -0,0 +1,60 @@
+import { Contract } from "../../../typings/contract";
+import { Embark } from "../../../typings/embark";
+import { Solidity } from "./contractLanguage/solidity";
+import { Description } from "./description";
+import { React } from "./framework/react";
+import { ContractLanguage, Framework, Options } from "./options";
+
+export default class Scaffolding {
+
+  constructor(private embark: Embark, private options: any) {
+    this.embark.events.setCommandHandler("scaffolding:generate:contract", this.generateContract.bind(this));
+    this.embark.events.setCommandHandler("scaffolding:generate:ui", this.generateUi.bind(this));
+  }
+
+  private readonly contractLanguageStrategy = (description: Description, options: Options) => {
+    switch (options.contractLanguage) {
+      case ContractLanguage.Solidity: {
+        return new Solidity(this.embark, description, options);
+      }
+    }
+  }
+
+  private readonly frameworkStrategy = (description: Description, contracts: Contract[], options: Options) => {
+    switch (options.framework) {
+      case Framework.React: {
+        return new React(this.embark, description, contracts, options);
```

Comment:
Seeing how this is used now, maybe we can call this `ReactFrameworkStrategy` or something like that instead.